components/scripts/lib/paths.sh

```diff
@@ -47,15 +47,9 @@ relative_path() {
 }
 
 relative_lib_path() {
-  #shellcheck disable=SC2164  # We will handle the error when we try to invoke the build
-  pushd "${project_dir}" > /dev/null 2>&1
```

Comment:
Unfortunately, I believe this change would break experiments 2, 3, and 5 in some cases because they also call this function, but outside of the context of the subproject.

More specifically, experiments 2, 3, and 5 call the `init_scripts_path` function, which calls this function. See [this in experiment 2](https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/main/components/scripts/gradle/02-validate-local-build-caching-same-location.sh#L116) for an example.

I believe a more reliable fix would be to [move the call to init_script_path in gradle.sh](https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/main/components/scripts/lib/gradle.sh#L15-L16) to be [before it pushds to the project dir](https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/main/components/scripts/lib/gradle.sh#L13) (move lines 15 and 16 to be before line 13).

That would be a quick-fix anyway. Another option is to update the experiments to not build the relative path to the additional init scripts in the experiment scripts and instead pass in the names of "extra" init scripts into the `invoke_gradle` function and calculate their relative paths there instead.  

Either fix would be fine with me.